subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/internal/CodeNarcInvoker.groovy

```diff
@@ -73,4 +82,12 @@ abstract class CodeNarcInvoker {
         }
     }
 
+    static void setLifecycleLogLevel(Object ant, String lifecycleLogLevel) {
+        ant?.builder?.project?.buildListeners?.each {
+            // We cannot use instanceof or getClass()==AntLoggingAdapter since they're in different class loaders
+            if (it.class.simpleName == 'AntLoggingAdapter') {
```

Comment:
Looks like there is no simpler way to do this.

We can improve it slightly, by replacing the check with
`it.class.name == AntLoggingAdapter.name`